### PR TITLE
background as variable

### PR DESCRIPTION
**Pull request by XuhuiZhou**
Closes # 

## 📑 Description

This change makes the Sotopia repo more general and suitable for more complex prompts and scenarios.

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed
- [x] Branch name follows `type/descript` (e.g. `feature/add-llm-agents`)
- [x] Ready for code review

### CODE REVIEW

File: sotopia/generation_utils/generate.py [hunk 1]
```diff
@@ -447,7 +447,7 @@ async def agenerate(
     output_parser: BaseOutputParser[OutputType],
     temperature: float = 0.7,
 ) -> OutputType:
-    input_variables = re.findall(r"{(.*?)}", template)
+    input_variables = re.findall(r"(?<!{){([^{}]+)}(?!})", template)
```
Comment (ProKil) [inline sotopia/generation_utils/generate.py:450]: Can you add some examples to the doc to illustrate why this is desirable? It seems that it ignores all nested curly parentheses. What is the reason behind this?

**Review by ProKil — CHANGES_REQUESTED** [review]
This looks good to me. Please add more explanation to the changes in `agenerate`

File: sotopia/generation_utils/generate.py (hunk 1 above)
Comment (XuhuiZhou) [reply sotopia/generation_utils/generate.py:450]: will write in the doc string

Comment (XuhuiZhou) [reply sotopia/generation_utils/generate.py]: updated (just the in-line explanation for now)
Will update the doc more in detail in the `dev` branch!

**Review by ProKil — APPROVED** [review]

**Pull request merged**